### PR TITLE
Fix test failing after porting feature to OData 8.0-beta

### PR DESCRIPTION
**Pull request by gathogojr**
Fixes test failing subsequent to porting feature to support passing of query options in request body
Related to #119

### CODE REVIEW

**Review by xuzhg — CHANGES_REQUESTED** [review]

File: test/Microsoft.AspNetCore.OData.E2E.Tests/QueryRequest/DollarQueryTests.cs [hunk 1]
```diff
@@ -16,14 +17,14 @@
 
 namespace Microsoft.AspNetCore.OData.E2E.Tests.Routing.QueryRequest
 {
-    public class DollarQueryTests : WebApiTestBase<DollarQueryTests>
```
[the remark this answers is not in the document]
Comment (gathogojr) [reply test/Microsoft.AspNetCore.OData.E2E.Tests/QueryRequest/DollarQueryTests.cs:19]: @xuzhg When working on the feature, a reviewer had requested for a test that verifies two things:
- that a request fails when the query options part of the URL exceeds a particular length, and
- that a request succeeds if you pass the same query options in the request body

That is what gave birth to the test in question. The challenge with `WebApiTestBase<T>` is that it makes use of `TestServer` and I couldn't figure out a way to impose request limits. Kestrel server on the other hand imposes the limits that make it possible to implement the requested test

That leaves us with the following options:
- get rid of the test altogether
- or, figure out a way to fail a request when the header violates configured limits when using `WebApiTestBase<T>`/`TestServer` - maybe we could use a middleware to achieve that?
- or, move this particular test to its own test class that derives from `WebHostTestBase<T>` class

File: test/Microsoft.AspNetCore.OData.E2E.Tests/QueryRequest/DollarQueryTests.cs [hunk 2]
```diff
@@ -227,52 +227,5 @@ public async Task ODataQueryOptionsInRequestBody_Empty()
             // Assert
             Assert.True(response.IsSuccessStatusCode);
         }
-
```
Comment (gathogojr) [inline test/Microsoft.AspNetCore.OData.E2E.Tests/QueryRequest/DollarQueryTests.cs:230]: Apart from confirming a known fact - that web server impose limitations on the request header size - this test doesn't add more value so its justifiable to drop it

File: src/Microsoft.AspNetCore.OData/ODataServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -109,7 +109,8 @@ private static IODataBuilder AddODataCore(this IServiceCollection services)
                 throw Error.ArgumentNull(nameof(services));
             }
 
-            services.TryAddSingleton<IODataQueryRequestParser, DefaultODataQueryRequestParser>();
+            services.TryAddEnumerable(
+                ServiceDescriptor.Singleton<IODataQueryRequestParser, DefaultODataQueryRequestParser>());
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/ODataServiceCollectionExtensions.cs:113]: @xuzhg Okay to register as Singleton or would Transient be preferable?

Comment (xuzhg) [reply src/Microsoft.AspNetCore.OData/ODataServiceCollectionExtensions.cs:113]: I think it should be singleton.

---
In reply to: [610428027](https://github.com/OData/AspNetCoreOData/pull/121#discussion_r610428027) [](ancestors = 610428027)

File: test/Microsoft.AspNetCore.OData.E2E.Tests/QueryRequest/DollarQueryTests.cs (hunk 2 above)
Comment (xuzhg) [reply test/Microsoft.AspNetCore.OData.E2E.Tests/QueryRequest/DollarQueryTests.cs:230]: would you please share a link for that?

---
In reply to: [610427377](https://github.com/OData/AspNetCoreOData/pull/121#discussion_r610427377) [](ancestors = 610427377)

**Review by xuzhg — APPROVED** [review]
:shipit:

File: test/Microsoft.AspNetCore.OData.Tests/PublicApi/Microsoft.AspNetCore.OData.PublicApi.NetCore31.bsl [hunk 4]
```diff
@@ -1369,7 +1369,7 @@ public class Microsoft.AspNetCore.OData.Query.ODataQueryOptions`1 : Microsoft.As
 }
 
 public class Microsoft.AspNetCore.OData.Query.ODataQueryRequestMiddleware {
-	public ODataQueryRequestMiddleware (Microsoft.AspNetCore.OData.Query.IODataQueryRequestParser queryRequestParser, Microsoft.AspNetCore.Http.RequestDelegate next)
```
Comment (xuzhg) [inline test/Microsoft.AspNetCore.OData.Tests/PublicApi/Microsoft.AspNetCore.OData.PublicApi.NetCore31.bsl:1372]: >queryRequestParser [](start = 95, length = 18)

It's weird,  It works without updating .NET 5 public API baseline?

File: test/Microsoft.AspNetCore.OData.E2E.Tests/QueryRequest/DollarQueryTests.cs (hunk 2 above)
Comment (gathogojr) [reply test/Microsoft.AspNetCore.OData.E2E.Tests/QueryRequest/DollarQueryTests.cs:230]: > would you please share a link for that?
> 
> In reply to: [610427377](https://github.com/OData/AspNetCoreOData/pull/121#discussion_r610427377) [](ancestors = 610427377)

@xuzhg http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#_Toc31360953

File: test/Microsoft.AspNetCore.OData.Tests/PublicApi/Microsoft.AspNetCore.OData.PublicApi.NetCore31.bsl (hunk 4 above)
Comment (gathogojr) [reply test/Microsoft.AspNetCore.OData.Tests/PublicApi/Microsoft.AspNetCore.OData.PublicApi.NetCore31.bsl:1372]: I updated the baseline for both .NET 3.1 and 5

**Pull request merged**